### PR TITLE
[16.0][MIG] barcodes_generator_product (from 14 to 16)

### PR DESCRIPTION
**Pull request by legalsylvain**
simple migration 
- bump version
- update fr translation
- add demo data
- use ``model_create_multi``
- improve product form display

product_template_manual_generation

### CODE REVIEW

**Comment by legalsylvain** [comment]
Hi. 
I have a problem with tests. (false positive).

- ``barcodes_generator_abstract`` load a fake model for tests.

- it has been introduced here https://github.com/OCA/stock-logistics-barcode/pull/340/commits/6cad8fdfc9c20f2e4ed3dd31dcd54710543fb4e0 by @CarlosRoca13, using the lib https://pypi.org/project/odoo-test-helper/

- when i load the view in the module ``barcodes_generator_product``, it raises an error : 
```
psycopg2.errors.UndefinedColumn: column res_users.barcode_rule_id does not exist
LINE 1: SELECT "res_users"."id" AS "id", "res_users"."barcode_rule_i...
```

However, the ``restore_registry`` is correctly called in the ``tearDownClass``.

anyone know how to solve that point ? 

Alternatively, we can remove the test in the ``barcodes_generator_abstract`` and the code will be tested in the dependent modules. ``_product``, ``_partner``, etc...

**Comment by legalsylvain** [comment]
CC : odoo-test-helper developpers : @simahawk @sebastienbeau
thanks !

**Comment by legalsylvain** [comment]
Friendly reminder : @CarlosRoca13, @simahawk, @sebastienbeau

Thanks !

**Comment by rousseldenis** [comment]
@legalsylvain It seems this one could solve the problem : https://github.com/OCA/odoo-test-helper/pull/22

**Comment by rousseldenis** [comment]
I've tested it successfully.

**Comment by legalsylvain** [comment]
Thanks for the investigation @rousseldenis. I wait for https://github.com/OCA/odoo-test-helper/pull/22 to be merged, and I'll retry after, to see if all is OK.

**Comment by rousseldenis** [comment]
> Thanks for the investigation @rousseldenis. I wait for [OCA/odoo-test-helper#22](https://github.com/OCA/odoo-test-helper/pull/22) to be merged, and I'll retry after, to see if all is OK.

@legalsylvain done :-)

**Comment by legalsylvain** [comment]
pre-commit errors unrelated.
16.0 should be fixed. (cf isort bug : https://github.com/PyCQA/isort/issues/2077)

**Comment by legalsylvain** [comment]
/ocabot rebase

since https://github.com/OCA/stock-logistics-barcode/pull/491 is merged.

**Review by aliciagaarzo — APPROVED** [review]

**Review by amarcosg — APPROVED** [review]

File: barcodes_generator_product/__manifest__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 {
     "name": "Generate Barcodes for Products",
     "summary": "Generate Barcodes for Products (Templates and Variants)",
-    "version": "14.0.1.1.0",
+    "version": "16.0.1.1.0",
```
Comment (pedrobaeza) [inline barcodes_generator_product/__manifest__.py]: 16.0.1.0.0

File: barcodes_generator_product/views/view_product_template.xml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    for="barcode"
+                    groups="barcodes_generator_abstract.generate_barcode"
+                    attrs="{'invisible': [('product_variant_count', '&gt;', 1)]}"
+                />
+                <div
+                    name="div_barcode"
+                    class="o_row"
+                    attrs="{'invisible': [('product_variant_count', '&gt;', 1)]}"
+                >
+                    <field
```
Comment (flalexg) [inline barcodes_generator_product/views/view_product_template.xml:65]: Checking the behaviour of the model I found a problem. When creating a new user, the group barcodes_generator_abstract.generate_barcode is disabled. If you enter to the Variant Products form view, the label of the field will not appear but the <field> yes, making the form view look wrong for the final user.

**Review by flalexg — CHANGES_REQUESTED** [review]

File: barcodes_generator_product/views/view_product_template.xml (hunk 2 above)
Comment (kikopeiro) [reply barcodes_generator_product/views/view_product_template.xml:65]: @legalsylvain The installation of this module dirty the view if the problem with the groups is not fixed. Could you check it?

Comment (kikopeiro) [reply barcodes_generator_product/views/view_product_template.xml:65]: Selección_758
Selección_759

Only is necesary remove 'groups' attribute on label for field 'barcode' inside product_product_view.xml and product_template_view.xml files.

Thanks!

**Comment by kikopeiro** [comment]
@legalsylvain  do you want us to make the proper changes or do you rather do it by yourself?

**Comment by legalsylvain** [comment]
Hi @kikopeiro. Sorry i missed your last comment. 
I have no time right now. If you provide a fix on my branch ,i'll review it quickly.

Regards.

**Comment by amarcosg** [comment]
@legalsylvain I have opened a [PR](https://github.com/legalsylvain/stock-logistics-barcode/pull/4) with the changes to your branch. That would be the fix we ask for.

Thanks!

**Comment by legalsylvain** [comment]
> @legalsylvain I have opened a https://github.com/legalsylvain/stock-logistics-barcode/pull/4 with the changes to your branch. That would be the fix we ask for.

Thanks! merged.

**Review by flalexg — APPROVED** [review]

**Comment by legalsylvain** [comment]
Could you wait https://github.com/legalsylvain/stock-logistics-barcode/pull/3 from @pierrickbrun before merging ?

**Review by hugo-cordoba — APPROVED** [review]
LGTM! (Code Review)

**Comment by hugo-cordoba** [comment]
Hello @pedrobaeza! Could you merge it??

Thanks! Regards!

File: barcodes_generator_product/__manifest__.py (hunk 1 above)
Comment (pedrobaeza) [reply barcodes_generator_product/__manifest__.py]: Still pending

Comment (hugo-cordoba) [reply barcodes_generator_product/__manifest__.py]: @legalsylvain could you change it please?

Comment (legalsylvain) [reply barcodes_generator_product/__manifest__.py]: Hi @pedrobaeza : don't know why github doesn't put a flag "obsolete" here but this diff is wrong.
The version is correct. see : https://github.com/OCA/stock-logistics-barcode/pull/450/files

this PR can be merged.

Comment (pedrobaeza) [reply barcodes_generator_product/__manifest__.py]: Something should happened in between, as if you go to the mentioned URL, the version is not the correct one.

Comment (legalsylvain) [reply barcodes_generator_product/__manifest__.py]: ?!? Well, when I go to that url (https://github.com/OCA/stock-logistics-barcode/pull/450/files) I see : 
```    "version": "16.0.1.1.0",```


Note : Ci is green and major version is checked by the CI AFAIK.

Comment (pedrobaeza) [reply barcodes_generator_product/__manifest__.py]: It should be `16.0.1.0.0`, which is the original claim.

Comment (legalsylvain) [reply barcodes_generator_product/__manifest__.py]: ow sorry for the noice.
Fixed.

**Pull request merged**